numba/tests/test_parfors.py

```diff
@@ -1383,6 +1383,22 @@ def test_impl(n):
         self.check(test_impl, n)
         self.assertEqual(countNonParforArrayAccesses(test_impl, (types.intp,)), 0)
 
+    @skip_parfors_unsupported
+    def test_parfor_array_access_lower_slice(self):
+        for ts in [slice(1, 3, None), slice(2, None, None), slice(None, 2, -1),
+                   slice(None, None, None), slice(None, None, -2)]:
+
+            def test_impl(n):
+                X = np.ones((n, 3))
+                y = 0
+                for i in numba.prange(n):
+                    y += X[i, ts].sum()
```

Comment:
What I thought the PR was about was if the array and the slice were defined outside that you could bake in the known value of that expression (although it seems like we'd want to do that early and not in lowering).  In this parallel test case, the array is inside and so that ability to replace with the known value is lost?  This may also be a useful test but perhaps add another test with the array external.  Please correct my misunderstanding if these comments are silly and I'm not understanding the point of this PR.

Reply:
I just added another test with the `X` array defined outside of the compiled function.  Is that closer to what you were thinking?